### ISSUE TITLE
[mqtt] Exception occurred while initializing handler of thing mqtt broker

### ISSUE THREAD

**Issue by EnziinSystem** [openhab/openhab-addons#13356]
My environment:

- Debian 10 run on BeagleBoard-X15
- Java: Runtime Environment Zulu11.56+19-CA (build 11.0.15+10-LTS)
- OpenHAB 3.4.0 Build 2986
- mosquitto version 2.0.12

I install and configure and it works well for some weeks.

Today, the MQTT broker thing stop working with the error message:

```
2022-09-05 04:11:59.128 [ERROR] [core.thing.internal.ThingManagerImpl] - Exception occurred while initializing handler of thing 'mqtt:broker:cd4df6f8f9': protocol
java.lang.NoSuchFieldError: protocol
	at org.openhab.binding.mqtt.handler.BrokerHandler.createBrokerConnection(BrokerHandler.java:203) ~[?:?]
	at org.openhab.binding.mqtt.handler.BrokerHandler.initialize(BrokerHandler.java:238) ~[?:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:?]
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
	at java.lang.reflect.Method.invoke(Method.java:566) ~[?:?]
	at org.openhab.core.internal.common.AbstractInvocationHandler.invokeDirect(AbstractInvocationHandler.java:154) [bundleFile:?]
	at org.openhab.core.internal.common.Invocation.call(Invocation.java:52) [bundleFile:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:829) [?:?]
```
![P1](https://user-images.githubusercontent.com/25495466/188360951-04a40741-5d68-4123-8974-e0b3e2b9bb6f.png)
![P2](https://user-images.githubusercontent.com/25495466/188360964-1d3d84d6-a220-4c01-b4b5-7509670f097f.png)

Check healthy mosquitto service, it works.

```
sudo systemctl status mosquitto.service
* mosquitto.service - Mosquitto MQTT Broker
   Loaded: loaded (/lib/systemd/system/mosquitto.service; enabled; vendor preset: enabled)
   Active: active (running) since Mon 2022-09-05 03:55:52 UTC; 42min ago
     Docs: man:mosquitto.conf(5)
           man:mosquitto(8)
  Process: 606 ExecStartPre=/bin/mkdir -m 740 -p /var/log/mosquitto (code=exited, status=0/SUCCESS)
  Process: 621 ExecStartPre=/bin/chown mosquitto /var/log/mosquitto (code=exited, status=0/SUCCESS)
  Process: 625 ExecStartPre=/bin/mkdir -m 740 -p /run/mosquitto (code=exited, status=0/SUCCESS)
  Process: 627 ExecStartPre=/bin/chown mosquitto /run/mosquitto (code=exited, status=0/SUCCESS)
 Main PID: 628 (mosquitto)
    Tasks: 1 (limit: 3370)
   Memory: 1.9M
   CGroup: /system.slice/mosquitto.service
           `-628 /usr/sbin/mosquitto -c /etc/mosquitto/mosquitto.conf
```

I re-install the MQTT broker but the error still while I can control devices via zigbee2mqtt and view topics by MQTT Exprorer

**Comment by wborn** [comment]
Does it work if you recreate the broker thing?

**Comment by EnziinSystem** [comment]
I re-install the MQTT broker but the error still.
After I downgrade to 3.3.0 and install everything then it works.

**Comment by wborn** [comment]
No I mean the Thing in openHAB, not the actual broker. 😉

**Comment by jimtng** [comment]
I am seeing the same error. I was using a pre-M1 snapshot and now updated to the latest snapshot build #3039 and the error is still there. I'm using file / text config.

**Comment by jimtng** [comment]
I encountered this problem just now whilst running a pre-M1 3.4 snapshot. I think it was triggered by an addon install, i.e. I added an addon to my addons.cfg.

It was resolved by updating to the latest snapshot #3088 and restarting my docker with `docker-compose up -d`.

**Comment by wborn** [comment]
You probably run into this issue if you run an old openHAB snaphot that has a core without the changes of https://github.com/openhab/openhab-core/pull/3068. If you use the online repo for installing add-ons, you'll always install the latest add-on snapshot version. So if you then install the MQTT add-on that has https://github.com/openhab/openhab-addons/pull/13303 you'll get the exception. So upgrading openHAB to a more recent snapshot version should fix it.
